packages/datadog-plugin-connect/test/index.spec.js

```diff
@@ -504,82 +505,234 @@ describe('Plugin', () => {
+                  })
+                  .catch(done)
+              })
+            })
           })
 
-          getPort().then(port => {
-            agent
-              .use(traces => {
-                const spans = sort(traces[0])
+          it('should do automatic instrumentation on app routes', done => {
+            const app = connect()
 
-                expect(spans[0]).to.have.property('error', 1)
+            app.use('/user', (req, res) => {
+              res.statusCode = 200
+              res.end()
+            })
+
+            getPort().then(port => {
+              agent
+                .use(traces => {
+                  const spans = sort(traces[0])
+
+                  expect(spans[0]).to.have.property('service', 'custom')
+                  expect(spans[0]).to.have.property('type', 'web')
+                  expect(spans[0]).to.have.property('resource', 'GET /user')
+                  expect(spans[0].meta).to.have.property('span.kind', 'server')
+                  expect(spans[0].meta).to.have.property('http.url', `http://localhost:${port}/user`)
+                  expect(spans[0].meta).to.have.property('http.method', 'GET')
+                  expect(spans[0].meta).to.have.property('http.status_code', '200')
+                })
+                .then(done)
+                .catch(done)
+
+              appListener = http.createServer(app).listen(port, 'localhost', () => {
+                axios
+                  .get(`http://localhost:${port}/user`)
+                  .catch(done)
               })
-              .then(done)
-              .catch(done)
+            })
+          })
 
-            appListener = http.createServer(app).listen(port, 'localhost', () => {
-              axios
-                .get(`http://localhost:${port}/user`, {
-                  validateStatus: status => status === 400
+          it('should not do automatic instrumentation on middleware', done => {
+            const app = connect()
+
+            app.use(function named (req, res, next) { next() })
+            app.use('/app/user', (req, res) => res.end())
+
+            getPort().then(port => {
+              agent
+                .use(traces => {
+                  const spans = sort(traces[0])
+
+                  expect(spans).to.have.length(1)
+                  expect(spans[0]).to.have.property('resource', 'GET /app/user')
+                  expect(spans[0]).to.have.property('name', 'connect.request')
                 })
+                .then(done)
                 .catch(done)
+
+              appListener = http.createServer(app).listen(port, 'localhost', () => {
+                axios
+                  .get(`http://localhost:${port}/app/user/1`)
+                  .catch(done)
+              })
             })
           })
-        })
 
-        it('should include specified headers in metadata', done => {
-          const app = connect()
+          it('should not activate a scope per middleware', done => {
```

Comment:
A scope should still be activated, but it should be the one of the request instead.

Reply:
Thank you for highlighting this. I've updated the underlying middleware instrumentation so that it now both leverages `web.reactivate` for the request span, and also captures any errors thrown by the middleware, and updated the tests accordingly to match this behavior (that the middleware is exposed to the request scope, not their own scope and not a null scope) . lmk what you think